### PR TITLE
Update Dockerfile and docker-compose files

### PR DESCRIPTION
**Pull request by duanehutchins**
* Consolidate Dockerfile commands for improved caching and performance
* Change Mongodb to use a named Docker volume
* Update docker-compose to start server without calling trade --paper
* Update docker-compose-windows to match UNIX they are now the same

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
     image: mongo:latest
     volumes:
-      - ./data/db:/data/db
+      - data:/data/db
     command: mongod --smallfiles --bind_ip=0.0.0.0 --logpath=/dev/null
     ports:
       - 27017:27017
 
+volumes:
+  data: {}
```
Comment (DeviaVir) [inline docker-compose.yml:31]: I'd  actually argue against this, as I'd prefer to have the data local in the same directory. Thoughts?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,31 +1,35 @@
...
-      - ./conf.js:/app/conf.js
-      - ./extensions:/app/extensions
-      - ./simulations:/app/simulations
+      - '.:/app'
     links:
       - mongodb
-    command: [ "trade", "--paper" ]
+    working_dir: /app
+    entrypoint: sh -c
+    command: ["./startup.sh"]
```
Comment (DeviaVir) [inline docker-compose.yml:13]: Why not `trade --paper`?

File: docker-compose.yml [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
 services:
-
   server:
     image: deviavir/zenbot:unstable
+    # build: .
     volumes:
-      - ./conf.js:/app/conf.js
-      - ./extensions:/app/extensions
-      - ./simulations:/app/simulations
+      - '.:/app'
```
Comment (DeviaVir) [inline docker-compose.yml:8]: I don't think this is a good idea, we're installing `node_modules` in the Dockerfile and this looks like it overwrites that

**Review by DeviaVir — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 3 above)
Comment (duanehutchins) [reply docker-compose.yml:8]: Ah, yes, you're right. I'll come up with a different solution.

File: docker-compose.yml (hunk 2 above)
Comment (duanehutchins) [reply docker-compose.yml:13]: If trade --paper fails, the container won't start. Also, sometimes I don't want it running the trade, so this gives me the ability to trigger it myself or not.

File: docker-compose.yml (hunk 1 above)
Comment (duanehutchins) [reply docker-compose.yml:31]: The benefit of a named volume gives me the ability to access the volume from another heroku container. It also makes it a bit more protected from accidental deletions when a user deletes his directory.

Comment (DeviaVir) [reply docker-compose.yml:31]: I understand the heroku use case, but I can't really get behind the accidental deletion: if I delete the zenbot dir I would really also want to delete all DB data, and not have that linger somewhere invisible to me.

**Pull request closed without merging**